modules/dxp/apps/segments/segments-asah-connector/src/test/java/com/liferay/segments/asah/connector/internal/messaging/IndividualSegmentsMessageListenerTest.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ *
+ *
+ *
+ */
+
+package com.liferay.segments.asah.connector.internal.messaging;
+
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.messaging.Message;
+import com.liferay.portal.kernel.test.ReflectionTestUtil;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+
+/**
+ * @author David Arques
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class IndividualSegmentsMessageListenerTest {
+
+	@Before
+	public void setUp() {
+		ReflectionTestUtil.setFieldValue(
+			_individualSegmentsMessageListener, "_individualSegmentsChecker",
+			_individualSegmentsChecker);
+	}
+
+	@Test
+	public void testDoReceive() throws Exception {
+		Message message = new Message();
+
+		message.put("companyId", RandomTestUtil.randomLong());
+		message.put("userId", RandomTestUtil.randomString());
+
+		_individualSegmentsMessageListener.doReceive(message);
+
+		Mockito.verify(
+			_individualSegmentsChecker, Mockito.times(1)
+		).checkIndividualSegments(
+			message.getLong("companyId"), message.getString("userId")
+		);
+	}
+
+	@Test
+	public void testDoReceiveWithEmptyAcClientUserId() throws Exception {
+		Message message = new Message();
+
+		message.put("userId", StringPool.BLANK);
+
+		_individualSegmentsMessageListener.doReceive(message);
+
+		Mockito.verify(
```

Comment:
We should not create test like: 

`Mockito.verify(
			_individualSegmentsChecker, Mockito.times(1)
		).checkIndividualSegments(
			message.getLong("companyId"), message.getString("userId")
		);`

This test is highly coupled to the implementation. here there is a nice chapter that explains best practices with Mocks in unit testing: https://livebook.manning.com/book/unit-testing/chapter-9/v-3/5

Reply:
In general, a test always should follow the steps:

1. Given an initial condition (this step is optional)
2. When something happens
3. A result is expected.